src/composables/transactionsList.ts

```diff
@@ -0,0 +1,229 @@
+import { useStore } from 'vuex';
+import { orderBy, uniqBy } from 'lodash-es';
+import { computed, ref } from 'vue';
+import { Encoded } from '@aeternity/aepp-sdk-13';
+import { ITransaction, ITransactionsState } from '../types';
+import { useSdk13 } from './sdk13';
+import { useSdk } from './sdk';
+import { useMiddleware } from './middleware';
+import { AEX9_TRANSFER_EVENT } from '../popup/utils';
+import { useAccounts } from './accounts';
+
+interface IFetchTransactionsParams {
+  limit: number,
+  recent: boolean,
+  address: Encoded.AccountAddress,
+  multipleAccounts?: boolean
+}
+
+const transactions = ref<ITransactionsState[]>([]);
+
+function getTransactionsByAccountAddress(address: Encoded.AccountAddress) {
+  return transactions.value.find((transactionState) => (
+    transactionState.accountAddress === address
+  ));
+}
+
+function getTransactionStateIndexByAccountAddress(address: Encoded.AccountAddress) {
+  return transactions.value.findIndex((transactionState) => (
+    transactionState.accountAddress === address
+  ));
+}
+function getOrInitTransactionStateForAccount(address :Encoded.AccountAddress): ITransactionsState {
+  const accountTransactionState = getTransactionsByAccountAddress(address);
+
+  if (accountTransactionState) {
+    return accountTransactionState;
+  }
+
+  const newAccountTransactionState: ITransactionsState = {
+    accountAddress: address,
+    loaded: [],
+    nextPageUrl: '',
+    pending: {},
+    tipWithdrawnTransactions: [],
+  };
+
+  transactions.value = [...transactions.value, newAccountTransactionState];
+
+  return newAccountTransactionState;
+}
+
+function updateTransactionByAccount(address: Encoded.AccountAddress, transaction: ITransaction) {
```

Comment:
I'm not a fan of the names `doSthForAccount`, `doSthFromAccount`,`doSthByAccount`. What do you think of naming it `doAccountSth` - in this case `updateAccountTransactions`?